### DISCUSSION TITLE
Support code blocks in Aside titles

### DISCUSSION THREAD

**Discussion by theoephraim** [withastro/starlight#1915] [Feature Requests]
### What version of `starlight` are you using?

0.23.1

### What is your idea?

The aside component when used from markdown should support code

```mdx 
Regular text with `code in it` works fine

## Normal titles `also are fine`

:::note[But `code` in titles like this does not]
Again in here, `some code` works
:::
```

### Why is this feature necessary?

Plenty of cases where an aside is some warning / note about code, and it's helpful to highlight the function name or whatever you are writing as code. Regardless is breaks expectations when code in similar places just next to the aside do work fine.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by essential-randomness** [comment]
I did some research of this and I think [this discussion on the remarkjs repo](https://github.com/orgs/remarkjs/discussions/725) should be of help. I'm also willing to attempt a fix in a PR.

**Reply by delucis** [reply]
Hi @essential-randomness! If it seems straightforward, would be happy to look at a PR for this!

**Comment by delucis** [comment]
Linking https://github.com/withastro/docs/issues/8711 here which reported a need for this also in Astro’s Docs.

**Comment by essential-randomness** [comment]
[Opened a PR for this](https://github.com/withastro/starlight/pull/2126), the solution ended up being simpler than I thought cause the markdown in the aside title was already being parsed correctly, we just needed to change the logic to use the whole mdast node for it.

**Comment by HiDeoo** [comment]
Starting with Starlight [`v0.25.2`](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.25.2), support for markdown formatting in aside titles is now available thanks to @essential-randomness 🙌
